opm/parser/eclipse/EclipseState/Schedule/Well.hpp

```diff
@@ -95,7 +95,7 @@ namespace Opm {
         void addWELSPECS(const DeckRecord& deckRecord);
 
         void addConnections(size_t time_step, const std::vector< Connection >& );
-        void addConnectionSet(size_t time_step, ConnectionSet );
+        void addConnectionSet(size_t time_step, std::shared_ptr<ConnectionSet> );
```

Comment:
Even though this PR is still in WIP and may be subject to changes; I wonder why is there a need to add shared pointers?

Reply:
Well, there is no *need* to use shared pointers, but these objects will go in a `DynamicState<>` object and the use of pointers will significantly reduce copying there; the same pattern is used for `WTESTConfig`and `VFP`tables where it has worked quite well.

However - as you mention this *is* WIP and the gole is that the use of shared pointers should be invisible in the public api.